mino/minogrpc/mod.go

```diff
@@ -202,38 +212,46 @@ func NewMinogrpc(listen net.Addr, public *url.URL, router router.Router, opts ..
 		return nil, xerrors.Errorf("overlay: %v", err)
 	}
 
-	chainBuf := o.GetCertificateChain()
-	certs, err := x509.ParseCertificates(chainBuf)
-	if err != nil {
-		socket.Close()
-		return nil, xerrors.Errorf("failed to parse chain: %v", err)
-	}
-
-	certsBuf := make([][]byte, len(certs))
-	for i, c := range certs {
-		certsBuf[i] = c.Raw
-	}
-
-	creds := credentials.NewTLS(&tls.Config{
-		Certificates: []tls.Certificate{{
-			Certificate: certsBuf,
-			Leaf:        certs[0],
-			PrivateKey:  o.secret,
-		}},
-		MinVersion: tls.VersionTLS12,
-	})
-
 	dialAddr := o.myAddr.GetDialAddress()
 	tracer, err := getTracerForAddr(dialAddr)
 	if err != nil {
 		return nil, xerrors.Errorf("failed to get tracer for addr %s: %v", dialAddr, err)
 	}
 
-	server := grpc.NewServer(
-		grpc.Creds(creds),
+	srvOpts := []grpc.ServerOption{
 		grpc.UnaryInterceptor(otgrpc.OpenTracingServerInterceptor(tracer, otgrpc.SpanDecorator(decorateServerTrace))),
 		grpc.StreamInterceptor(otgrpc.OpenTracingStreamServerInterceptor(tracer, otgrpc.SpanDecorator(decorateServerTrace))),
-	)
+	}
+
+	if !tmpl.noTLS {
+		chainBuf := o.GetCertificateChain()
+		certs, err := x509.ParseCertificates(chainBuf)
+		if err != nil {
+			socket.Close()
+			return nil, xerrors.Errorf("failed to parse chain: %v", err)
+		}
+
+		certsBuf := make([][]byte, len(certs))
+		for i, c := range certs {
+			certsBuf[i] = c.Raw
+		}
+
+		creds := credentials.NewTLS(&tls.Config{
+			Certificates: []tls.Certificate{{
+				Certificate: certsBuf,
+				Leaf:        certs[0],
+				PrivateKey:  o.secret,
+			}},
+			MinVersion: tls.VersionTLS12,
+		})
+
+		srvOpts = append(srvOpts, grpc.Creds(creds))
+	} else {
+		dela.Logger.Warn().Msg("⚠️ running in insecure mode, you should not " +
+			"publicly expose the node's socket")
+	}
```

Comment:
since this is very short, it might make sense to switch the `if` and `else` clauses, to improve readability 
i.e. 
```go
if !tmpl.useTLS {
  dela.Logger.Warn().Msg(...)
} else {
  // do all the other long stuff
}